app/controllers/grades_controller.rb

```diff
@@ -233,59 +233,49 @@ def calculate_all_penalties(assignment_id)
   end
 
   def calculate_penatly_attributes(_participant)
-    penalty_attr1 = {deadline_type_id: 1, participant_id: @participant.id, penalty_points: penalties[:submission]}
-    CalculatedPenalty.create(penalty_attr1)
-
-    penalty_attr2 = {deadline_type_id: 2, participant_id: @participant.id, penalty_points: penalties[:review]}
-    CalculatedPenalty.create(penalty_attr2)
-
-    penalty_attr3 = {deadline_type_id: 5, participant_id: @participant.id, penalty_points: penalties[:meta_review]}
-    CalculatedPenalty.create(penalty_attr3)
+    deadline_type_id = [1, 2, 5]
```

Comment:
Good refactoring!
For deadline_type_id, it will be better to refer to [this file](https://github.com/jerry-shijieli/expertiza/blob/d76ee028113474421d281f8c29a98e7587924e7f/app/helpers/deadline_helper.rb#L4) and use `DeadlineHelper::DEALINE_TYPE_SUBMISSION` instead of `1` and so on. It makes the code more readable.